### PR TITLE
Headless/dipu bala/bc 7/branch/branch request history

### PR DESCRIPTION
**Pull request by DipuBala**
# Description

(Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.)

## Type of change

(Please delete options that are not relevant.)

- [x] Link to a **GH Issue** or **https://pm23.atlassian.net/browse/BC-722** task here for reference.
- [x] Bug fix
- [ ] New feature

# How Has This Been Tested?

(Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration)

- [x] Tested by myself
- [ ] Unit test

# Developer Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

# Review check list:

- [x] Separation of Concerns followed (Reviewer name/NaN)
- [x] Code is in sync with existing code patterns/technologies. (Reviewer name/NaN)
- [ ] Are there any obvious optimizations that will improve performance? (Reviewer name/NaN)
- [ ] Can any logging or debugging code be removed? (Reviewer name/NaN)
- [ ] Are all data inputs checked (for the correct type, length, format, and range) and encoded? (Reviewer name/NaN)
- [ ] Are invalid parameter values handled? (Reviewer name/NaN)
- [ ] Is error handling done the correct way? (Reviewer name/NaN)
- [ ] Are all the edge cases (null, 0, negative) handled properly? (Reviewer name/NaN)

### CODE REVIEW

File: packages/headless/src/modules/branch/services/index.ts [hunk 1]
```diff
@@ -1,82 +1,189 @@
...
+
+    const history = {
+      id: randomUUID(),
+      branchName: branch.name,
+      branchURL: branchBody.url,
+      actions: [
+        {
+          user: {
+            id: '123',
+            email: 'admin@gmail.com',
```
Comment (Anjarul-Islam) [inline packages/headless/src/modules/branch/services/index.ts:75]: why hard coded admin?

File: packages/headless/src/modules/branch/services/index.ts [hunk 2]
```diff
@@ -1,82 +1,189 @@
...
-          errors: null,
-          code: HttpStatus.BAD_REQUEST,
-        };
+    if (status === StoreBranchStatus.APPROVED) {
+      const Actions: any = [
+        {
+          user: {
+            id: '123',
+            email: 'admin@gmail.com',
+            type: UserTypeEnum.ADMIN,
```
Comment (Anjarul-Islam) [inline packages/headless/src/modules/branch/services/index.ts:150]: why hard-coded admin?

File: packages/headless/src/modules/branch/services/index.ts [hunk 3]
```diff
@@ -1,82 +1,189 @@
...
         HttpStatus.OK,
       );
-    }
+    } else if (status === StoreBranchStatus.REJECTED) {
+      const Actions: any = [
+        {
+          user: {
+            id: '123',
+            email: 'admin@gmail.com',
+            type: UserTypeEnum.ADMIN,
```
Comment (Anjarul-Islam) [inline packages/headless/src/modules/branch/services/index.ts:169]: hard-coded admin?

File: packages/headless/src/modules/tmp-store-branch/services/index.ts [hunk 4]
```diff
@@ -33,35 +42,59 @@ export class StoreBranchService {
...
+      id: randomUUID(),
+      branchName: storeBranch.name,
+      branchURL: branchBody.url,
+      actions: [
+        {
+          user: {
+            id: '123',
+            email: 'admin@gmail.com',
+            type: UserTypeEnum.MERCHANT,
+          },
```
Comment (Anjarul-Islam) [inline packages/headless/src/modules/tmp-store-branch/services/index.ts:85]: hard-coded admin?

File: packages/headless/src/modules/branch/services/index.ts (hunk 1 above)
Comment (DipuBala) [reply packages/headless/src/modules/branch/services/index.ts:75]: had some dependency on store admin and super admin authentication. I will fix this issue after getting  that module .

**Review by Anjarul-Islam — APPROVED** [review]

**Pull request merged**